File: 2011-strangeloop-resources/samples/cache_factory.rb
require 'java'

class CacheFactory

  include org.infinispan.Cache
  include org.infinispan.config.Configuration
  include org.infinispan.loaders.file.FileCacheStore
  include org.infinispan.loaders.file.DefaultCacheManager

  def initialize
    @config = Configuration.new
    @store  = FileCacheStore.new

    # Configure the cache
    @store.purge_on_startup false
    @config.fluent.loaders.add_cache_loader( @store ).indexing.index_local_only( true ).add_property( 'indexing', 'in memory' )
    
    # Create the cache and return it
    @manager = DefaultCacheManager.new( @config.build )
  end

  def get_cache
    @manager.get_cache
  end

end

File: 2011-strangeloop-resources/samples/CacheFactory.java
import org.infinispan.Cache;
import org.infinispan.config.Configuration;
import org.infinispan.loaders.file.FileCacheStore;
import org.infinispan.loaders.file.DefaultCacheManager;


public class CacheFactory {

  private Configuration config;
  private FileCacheStoreConfig store;
  private DefaultCacheManager manager;

  public CacheFactory() {
    config = new Configuration();
    store  = new FileCacheStoreConfig();
    
    // Configure the cache
    store.purgeOnStartup( false );
    config.fluent().loaders.addCacheLoader( store ).indexing().indexLocalOnly( true ).addProperty("indexing", "in memory");

    // Create the cache and return it
    manager = new DefaultCacheManager( config.build() );
  }

  public Cache getCache() {
    return manager.getCache();
  }
}
